### ISSUE TITLE
spack: Issues

### ISSUE THREAD

**Issue by kdy1** [swc-project/swc#841]
# Tasks

 - [ ] Handle require
 - [ ] Handle dynamic import

 - [ ] Provide webpack-like plugin / loader /resolver api

This is really a show-stopper. I'm going to publish first version without support for plugins. This requires one of the below, as spack uses all cpu cores.

 - https://github.com/neon-bindings/neon/issues/73
 - https://github.com/infinyon/node-bindgen/issues/37

 - [ ] Implement minifier (#826)

**Comment by jakajancar** [comment]
Hey, is spack on crates.io yet?
All I'm seeing is the `spack` crate, but the docs just point to https://swc-project.github.io/rustdoc/swc/.

(And on the side note, is there an actual crate that exposes `swc::` that is documented on the above link, or do you need to include individual components?)

**Comment by kdy1** [comment]
@jakajancar  Sorry for late response. 
I don't think spack as a crate is mature enough to be published. It's api will be changed frequently, just like the swc crate.

Managing multi crate repository is a big pain if some of those are published...

**Comment by kdy1** [comment]
@jakajancar Bundler is now on crates.io
See: https://crates.io/crates/swc_bundler

**Comment by jrop** [comment]
@kdy1 Is there a recommended way to invoke `spack` at build-time (e.g., from a `build.rs` script)?  That is all I really want from a spack-API.

**Comment by kdy1** [comment]
@jrop You can use `Process`

**Comment by jrop** [comment]
@kdy1 I was hoping to avoid a dependency on NodeJS

**Comment by ScottAwesome** [comment]
> @kdy1 I was hoping to avoid a dependency on NodeJS

I don't think you can, since SWC uses [neon](https://github.com/neon-bindings/neon) for parts of the parser and interop. One way or another you'll need to have node installed if I understand this correctly.

**Comment by kdy1** [comment]
It's possible, but I'm not sure if it worths.

**Comment by Brooooooklyn** [comment]
> I don't think you can, since SWC uses neon for parts of the parser and interop.

@ScottAwesome no, swc uses [napi-rs](https://napi.rs)
